docs/DEVELOP.md

```diff
@@ -235,6 +264,71 @@ Please read the [MDN docs](https://developer.mozilla.org/en-US/docs/Mozilla/Add-
 
 Note that, we only support [manifest version 3](https://developer.chrome.com/docs/extensions/develop/migrate/what-is-mv3), and not version 2.
 
+### Payment mechanism
+
+1. The content script finds the valid `<link rel="monetization">` elements and sends a message to the background.
+2. The background then sets up a payment session for this element - via `Monetization` (singleton for browser instance) → `PaymentManager` (for tab) → `PaymentStream` (for frame) → `PaymentSession` (for element).
+3. Once a payment is made, the `PaymentSession` passes on an event to the website (via `polyfill.ts`), and in turn a `MonetizationEvent` is emitted.
+
+#### `minSendAmount`
+
+When a `PaymentSession` is created, we find a `minSendAmount` before we try to make any payment. This value is used in figuring out the amounts and timings for the payments for given link element. The `minSendAmount` is obtained by creating a quote with particular amount and seeing if the quote is accepted at that amount.
+
+The `minSendAmount` is a property of user's connected wallet (currency/assetScale), the wallet for the given link element and associated payment fees at that time. For a browsing session (which _resets_ on navigation), we typically find this amount only once. But if future payments fail, we may need to adjust the `minSendAmount` again.
+
+We make payments only in multiples of `minSendAmount`. This ensures some fairness for the sender and receivers - as the amounts higher than in-multiple of `minSendAmount` are absorbed by ASEs to maintain cross-currency liquidity for other transactions when user may send an amount less than multiple of `minSendAmount`.
+
+#### Open-payments flow
+
+To make a payment with Open Payments, following steps are taken (presuming we've the OutgoingPaymentGrant tokens available, i.e. user has connected the extension to their wallet):
+
+1. Create incoming payment:
+   1. Set up a non-interactive incoming payment grant for the receiving wallet address.
+   1. Create an incoming payment using this grant for the receiving wallet address.
+   1. Optionally, cancel the incoming payment grant, as we no longer need it (we already have the incoming payment). This prevents users wallet seeing _dangling_ grants in their wallet.
```

Comment:
@sidvishnoi Can you tell me more about this step for canceling the incoming payment grant? I'm only aware of completing an incoming payment (/incoming-payments/{id}/complete), which would tell the recipient's wallet that no more payments are coming.

Reply:
To explain, the incoming payment grant is for "creating" incoming payments. Once we've created our incoming payment, we have no further use of that grant anymore.